library/src/com/google/maps/android/kml/KmlContainer.java

```diff
@@ -0,0 +1,254 @@
+        return mPlacemarks;
+    }
+
+    /**
+     * Sets a property to be contained by the container
+     *
+     * @param propertyName  Name of the property, ie "name"
+     * @param propertyValue Value of the property, ie "Arizona"
+     */
+    public void setProperty(String propertyName, String propertyValue) {
+        mProperties.put(propertyName, propertyValue);
+    }
+
+    /**
+     * Sets a style to be contained by the container
+     *
+     * @param styleId Name or ID of the style
+     * @param style   KmlStyle object
+     */
+    public void setStyle(String styleId, KmlStyle style) {
+        mStyles.put(styleId, style);
+    }
+
+    /**
+     * @return A map of strings representing a style map, null if no style maps exist
+     */
+    /* package */ HashMap<String, String> getStyleMap() {
+        return mStyleMap;
+    }
+
+    /**
+     * @param styleMap Adds a map of strings representing a style map
+     */
+    /* package */ void setStyleMap(HashMap<String, String> styleMap) {
+        mStyleMap.putAll(styleMap);
+    }
+
+    /**
+     * Add a ground overlay for this container
+     *
+     * @param groundOverlay ground overlay to add
+     */
+    /* package */ void addGroundOverlay(KmlGroundOverlay groundOverlay) {
+        mGroundOverlays.put(groundOverlay, null);
+    }
+
+    /**
+     * Gets all of the ground overlays which were set in the container
+     *
+     * @return  A set of ground overlays
+     */
+    /* package */ HashMap<KmlGroundOverlay, GroundOverlay> getGroundOverlayHashMap() {
+        return mGroundOverlays;
+    }
+
+    /**
+     * Gets the value of a property based on the given key
+     *
+     * @param propertyName property key to find
+     * @return value of property found, null if key doesn't exist
+     */
+    public String getKmlProperty(String propertyName) {
+        return mProperties.get(propertyName);
+    }
+
+    /**
+     * Gets whether the container has any properties
+     *
+     * @return true if there are properties, false otherwise
+     */
+    public boolean hasKmlProperties() {
+        return mProperties.size() > 0;
+    }
+
+    /**
+     * Gets whether the given key exists in the properties
+     *
+     * @param keyValue property key to find
+     * @return true if key was found, false otherwise
+     */
+    public boolean hasKmlProperty(String keyValue) {
+        return mProperties.containsKey(keyValue);
+    }
+
+    /**
+     * Gets whether the container has containers
+     *
+     * @return true if there are containers, false otherwise
+     */
+    public boolean hasNestedKmlContainers() {
+        return mContainers.size() > 0;
+    }
+
+    /**
+     * Gets an iterable of nested KmlContainers
+     *
+     * @return iterable of KmlContainers
+     */
+    public Iterable<KmlContainer> getNestedKmlContainers() {
+        return mContainers;
+    }
+
+    /**
+     * Gets an iterable of the properties hashmap entries
```

Comment:
Drop the 'hashmap' part, unless it's part of the public interface. This allows you to change the implementation later.